jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -230,7 +230,10 @@ public long length()
      */
     public InputStream newInputStream() throws IOException
     {
-        return Files.newInputStream(getPath(), StandardOpenOption.READ);
+        Path path = getPath();
+        if (path == null)
+            return null;
```

Comment:
I would expect this to throw `IOException` not return null from the Javadoc. 
So I think the javadoc should be updated as well.